src/comodo/drakeSimulator/drakeSimulator.py

```diff
@@ -0,0 +1,243 @@
+        if self.visualize_robot_flag:
+            self.diagram.ForcedPublish(self.context)
+
+    def reset(self):
+        logging.error("Drake simulator reset not implemented, recreate drake_instance")
+        pass
+
+    def set_visualize_robot_flag(self, visualize_robot):
+        # pass meshcat to visualise the robot
+        self.visualize_robot_flag = visualize_robot
+        if self.visualize_robot_flag and not self.active_meshcat:
+            self.meshcat = StartMeshcat()
+            self.active_meshcat = True
+        pass
+
+    # does this need to simulator specific?
+    def set_base_pose_in_drake(self, xyz_rpy):
+        base_xyz_quat = np.zeros(7)
+        # order -- quaternion+xyz
+        base_xyz_quat[:4] = self.RPY_to_quat(xyz_rpy[3], xyz_rpy[4], xyz_rpy[5])
+        base_xyz_quat[4:] = xyz_rpy[:3]
+        self.qpos[:7] = base_xyz_quat
+
+    # set initial positions of the joints
+    def set_joint_vector_in_drake(self, pos):
+        self.qpos[7:] = pos
+        pass
+
+    def set_input(self, input):
+        # expose the acutation output port of the system to accept control
+        # inputs from the TSID and CentroidalMPC and step drake simulator
+        self.control_input_port.FixValue(self.context, input)
+        pass
+
+    def step(self, n_step=1, visualize=True):
+        # TODO: does nothing currently
+        # self.control_input_port.FixValue(self.context, self.joint_torques)
+        self.simulator.AdvanceTo(self.context.get_time() + n_step * self.time_step)
+        if visualize and self.visualize_robot_flag:
+            self.diagram.ForcedPublish(self.context)
+        pass
+
+    def step_with_motors(self, n_step, torque):
+        # implementation of motor level control inputs?
+        pass
+
+    def get_base(self):
+        # this is for the state output return
+        # quat + pos
+        base_pos = self.state_output_port.Eval(self.context)[:7]
+        # normalise unnormalised quaternion
+        base_quat_wxyz = base_pos[:4]
+        H_b = RigidTransform(
+            Quaternion(base_quat_wxyz / (np.linalg.norm(base_quat_wxyz))), base_pos[4:]
+        ).GetAsMatrix4()
+        return H_b
+
+    def get_base_velocity(self):
+        # this is for the state output return
+        base_vel = self.state_output_port.Eval(self.context)[self.nq : self.nq + 6]
+        # order -- linar_vel + angular_vel
+        return np.concatenate((base_vel[3:], base_vel[:3]))
+
+    # TODO: Change from get_state to get_joint_state?
+    def get_state(self):
+        # this is for the state output return
+        robot_state = self.state_output_port.Eval(self.context)
+        robot_pos = robot_state[: self.nq]
+        robot_vel = robot_state[self.nq : self.nq + self.nv]
+        # return joint state to be coherent with the MuJoCo API
+        return (
+            robot_pos[7:],
+            robot_vel[6:],
+            self.joint_torques,
+        )
+
+    def close(self):
```

Comment:
What should be this issue ? could you either document (with a warning) or remove it please ?

Reply:
I thin this is now resolved